frontend/ui/elements/page_overlap.lua

```diff
@@ -0,0 +1,103 @@
+local Device = require("device")
+local ReaderUI = require("apps/reader/readerui")
+local UIManager = require("ui/uimanager")
+local _ = require("gettext")
+local Screen = Device.screen
+local T = require("ffi/util").template
+
+local PageOverlap = {
+    text = _("Page overlap"),
+    sub_item_table = {
+        {
+            text_func = function()
+                local text = _("Page overlap")
+                if G_reader_settings:isTrue("page_overlap_enable") then
+                    text = text .. "   ★"
+                end
+                return text
+            end,
+            checked_func = function()
+                return ReaderUI.instance.view:isOverlapAllowed() and ReaderUI.instance.view.page_overlap_enable
+            end,
+            callback = function()
+                if ReaderUI.instance.view:isOverlapAllowed() then
+                    if ReaderUI.instance.view.page_overlap_enable then
+                        ReaderUI.instance.view.dim_area:clear()
+                    end
+                    ReaderUI.instance.view.page_overlap_enable = not ReaderUI.instance.view.page_overlap_enable
```

Comment:
(
You could may be make this easier to read when it's crowded like in there by using some of these at the start of the function:
```lua
local ui = ReaderUI.instance
local view = ReaderUI.instance.view
```
)